addOns/authhelper/src/main/java/org/zaproxy/addon/authhelper/AuthUtils.java

```diff
@@ -199,7 +198,7 @@ public void notifyMessageReceived(HttpMessage message) {
 
     private static long timeToWaitMs = TimeUnit.SECONDS.toMillis(5);
 
-    @Setter private static HistoryProvider historyProvider = new HistoryProvider();
+    private static HistoryProvider historyProvider = ExtensionAuthhelper.getHistoryProvider();
```

Comment:
Why remove the `@Setter` and then add a setter? Is that simply because of the JavaDoc? (The class is internal.)

Reply:
`ClientSideHandler` is still pending.

Reply:
Done